### ISSUE TITLE
Nuke `should`

### ISSUE THREAD

**Issue by ScottFreeCode** [mochajs/mocha#3035]
I am tired of running into this: https://travis-ci.org/mochajs/mocha/builds/281135003?utm_source=github_status&utm_medium=notification [To save others digging: these tests are fine, they pass on Node, they would pass in browser but we're using only expect.js in browser because there used to be no `should` compatibility there.]

We need to go through the remaining uses of `should` in our tests and change them to `expect`, then remove `should` from our mocha.opts and any fixtures, then remove `should` from package.json!

**Comment by ngeor** [comment]
I can pick this one up.

**Comment by ngeor** [comment]
I run into a problem with `diffs.spec.js`, which might be a breaking change. In short, using `expect` gives less verbose diffs compared with `should`.

With expect:

```
  1) diffs
       should display a diff for small strings:
     Error: expected 'foo rar baz' to equal 'foo bar baz'
      at Assertion.assert (node_modules/expect.js/index.js:96:13)
      at Assertion.be.Assertion.equal (node_modules/expect.js/index.js:216:10)
```

With should:

```
  1) diffs
       should display a diff for small strings:

      AssertionError: expected 'foo rar baz' to be 'foo bar baz'
      + expected - actual

      -foo rar baz
      +foo bar baz

      at Assertion.fail (node_modules/should/cjs/should.js:275:17)
      at Assertion.value (node_modules/should/cjs/should.js:356:19)
```



This is how `should` outputs errors (notice the diff):

```
9) diffs
       should work with objects:

      AssertionError: expected Object { name: 'tobi', species: 'ferret', color: 'white', age: 2 } to equal Object { name: 'loki', species: 'ferret', color: 'brown', age: 2 } (at name, A has 'tobi' and B has 'loki')
      + expected - actual

       {
         "age": 2
      -  "color": "white"
      -  "name": "tobi"
      +  "color": "brown"
      +  "name": "loki"
         "species": "ferret"
       }
      
      at Assertion.fail (node_modules/should/cjs/should.js:275:17)
      at Assertion.value (node_modules/should/cjs/should.js:356:19)
```

On the other hand, these tests are effectively testing the behavior of `should` so I don't know if mocha should have them here in the first place.

I'd like some feedback on how to tackle this one :-)

**Comment by ngeor** [comment]
PR created, waiting for feedback :)

**Comment by ScottFreeCode** [comment]
Fixed in #3042
